silver/tests/unit/test_invoice.py

```diff
@@ -88,3 +90,36 @@ def test_cancel_issued_invoice_with_related_proforma(self):
         proforma.invoice.save()
 
         assert proforma.invoice.state == proforma.state == Invoice.STATES.CANCELED
+
+    def _get_decimal_places(self, number):
+        return max(0, -number.as_tuple().exponent)
+
+    def test_invoice_total_decimal_points(self):
+        invoice_entries = DocumentEntryFactory.create_batch(3)
+        invoice = InvoiceFactory.create(invoice_entries=invoice_entries)
+
+        assert self._get_decimal_places(invoice.total) == 2
+
+    def test_invoice_total_before_tax_decimal_places(self):
+        invoice_entries = DocumentEntryFactory.create_batch(3)
+        invoice = InvoiceFactory.create(invoice_entries=invoice_entries)
+
+        invoice.sales_tax_percent = Decimal('20.00')
+
+        assert self._get_decimal_places(invoice.total_before_tax) == 2
```

Comment:
@bogdanpetrea since those tests failed to catch this bug till now, I'm with @zalmanu on this one.